workflows/gen_crates.py

```diff
@@ -0,0 +1,66 @@
+import json
+import os
+import pathlib
+
+# pip install 'rocrate==0.4.0'
+from rocrate.rocrate import ROCrate
+
+THIS_DIR = pathlib.Path(__file__).absolute().parent
+
+ROOT = THIS_DIR / "sars-cov-2-variant-calling"
+
+GH_URL = "https://api.github.com"
+GH_RESOURCE = "repos/galaxyproject/iwc/actions/workflows/workflow_test.yml"
+
+PLANEMO_VERSION = ">=0.74.3"
+
+
+def get_wf_id(crate_dir):
+    ids = [_.name for _ in os.scandir(crate_dir) if _.name.endswith(".ga")]
+    assert len(ids) == 1
+    return ids[0]
+
+
+def get_planemo_id(crate_dir, wf_id):
+    tag, _ = os.path.splitext(wf_id)
+    planemo_id = f"{tag}-test.yml"
+    planemo_source = pathlib.Path(crate_dir) / planemo_id
+    assert planemo_source.is_file()
+    return planemo_id, planemo_source
+
+
+def main():
+    for entry in os.scandir(ROOT):
+        if not entry.is_dir():
+            continue
+        crate_dir = entry.path
+        wf_id = get_wf_id(crate_dir)
+        planemo_id, planemo_source = get_planemo_id(crate_dir, wf_id)
+        crate = ROCrate(gen_preview=False)
+        wf_source = pathlib.Path(crate_dir) / wf_id
+        with open(wf_source) as f:
+            code = json.load(f)
+        author = [_["name"] for _ in code["creator"]]
+        if len(author) == 1:
+            author = author[0]
+        workflow = crate.add_workflow(wf_source, wf_id, main=True,
+                                      lang="galaxy", gen_cwl=False)
+        workflow["name"] = code["name"]
+        crate.root_dataset["author"] = author
+        wf_url = f"https://github.com/iwc-workflows/{entry.name}"
```

Comment:
Can we make everything that is a namespace configurable ? I am hoping this can become a template for other organizations as well.

Reply:
Done in 53495ce. I'd never thought about turning this quick hack into a proper tool before, but now I'm starting to see it doesn't take much to make it do something useful. In the LM project we're currently thinking about ways to make workflow testing easier in general, so this could even be the beginning of something. It's starting to shape up now. Thanks for the insight!